### PR TITLE
Issue #13321: Kill mutation for SummaryJavaDocCheck-10

### PR DESCRIPTION
**Pull request by Kevin222004**
Issue #13321: Kill mutation for SummaryJavaDocCheck-10

-----

# Check :- 
https://checkstyle.org/checks/javadoc/summaryjavadoc.html#SummaryJavadoc

-----

# Mutation 
https://github.com/checkstyle/checkstyle/blob/aed4d3835de0da5e12c41e677e9a10be77b8a372/config/pitest-suppressions/pitest-javadoc-suppressions.xml#L975-L1000

-----

# Explaination
Test added

-----

# Regression :- 

-----

Diff Regression config: https://gist.githubusercontent.com/Kevin222004/61eaa69fe98585b6ee1882ca082362bc/raw/316ca02ce9fd290119d9bfc560f2b11e84a1b685/Summary.xml
Diff Regression projects: https://gist.githubusercontent.com/Kevin222004/21e3934e85f802e2fbd48af06d122364/raw/604256badd733d8568064f371d55657c04b00dfd/test-projects-2.properties
Report label: Regression-1

### CODE REVIEW

File: config/pitest-suppressions/pitest-javadoc-suppressions.xml [hunk 1]
```diff
@@ -972,32 +972,32 @@
...
-    <sourceFile>SummaryJavadocCheck.java</sourceFile>
-    <mutatedClass>com.puppycrawl.tools.checkstyle.checks.javadoc.SummaryJavadocCheck</mutatedClass>
-    <mutatedMethod>validateUntaggedSummary</mutatedMethod>
-    <mutator>org.pitest.mutationtest.engine.gregor.mutators.RemoveConditionalMutator_EQUAL_IF</mutator>
-    <description>removed conditional - replaced equality check with true</description>
-    <lineContent>else if (!period.isEmpty()) {</lineContent>
-  </mutation>
+
+
+
```
Comment (romani) [inline config/pitest-suppressions/pitest-javadoc-suppressions.xml:979]: are this empty lines to avoid conflicts ?

Comment (Kevin222004) [reply config/pitest-suppressions/pitest-javadoc-suppressions.xml:979]: Yes

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocTestForbiddenFragments.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+
+*/
+
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc.summaryjavadoc;
+
+public class InputSummaryJavadocTestForbiddenFragments {
+
+    /** // violation 'Forbidden summary fragment'
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocTestForbiddenFragments.java:14]: Move violation comment outside of javadoc multiline comment

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocEmptyPeriod.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc.summaryjavadoc;
+
+// ok
+public class InputSummaryJavadocEmptyPeriod {
+
+    /**
+     * checkStyle. check rule of CheckStyle.
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocEmptyPeriod.java:16]: Add some cases that supposed to match `CheckStyle$`

**Review by romani — CHANGES_REQUESTED** [review]
Items

**Comment by romani** [comment]
Please share diff report with empty period.
We need to make sure we keep report no violations on all projects in diff regression.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocTestForbiddenFragments.java (hunk 2 above)
Comment (Kevin222004) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocTestForbiddenFragments.java]: Done

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocEmptyPeriod.java (hunk 3 above)
Comment (Kevin222004) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocEmptyPeriod.java:16]: Done

**Comment by Kevin222004** [comment]
Github, generate report

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocEmptyPeriod.java (hunk 3 above)
Comment (romani) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocEmptyPeriod.java:16]: I meant such work in  first sentence.

Comment (Kevin222004) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocEmptyPeriod.java:16]: Please check

**Comment by Kevin222004** [comment]
failure not related to changes

**Comment by Kevin222004** [comment]
https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/b8fb668_2023045046/reports/diff/index.html I am sorry I think this happen becuase config fragment summary is `.*`  :)

**Comment by Kevin222004** [comment]
Github, generate report

**Comment by romani** [comment]
Conflict.
@Kevin222004 , please fix.

**Review by romani — APPROVED** [review]
Ok to merge

**Pull request merged**